python/dgl/base.py

```diff
@@ -50,6 +51,9 @@ def dgl_warning_format(message, category, filename, lineno, line=None):
 
 def dgl_warning(message, category=DGLWarning, stacklevel=2):
     """DGL warning wrapper that defaults to ``DGLWarning`` instead of ``UserWarning`` category."""
+    # Skip warning in unit test.
+    if "unittest" in sys.modules.keys():
```

Comment:
skip all warnings is too mighty...

Reply:
The problem is "did we ever care about warnings in unit test", I can move this to the item_sampler.py related to https://github.com/dmlc/dgl/pull/7047, and we can see whether the warning in unit test is helpful or just more trouble.

Reply:
I think we should care about warnings even in unit test in default as it reminds us we may call APIs in not-recommended way. For those we really don't care about it, we'd better capture warning explicitly right in the testcase which indicates: we know warning is thrown and we don't care about it. As the `timestamp` warning in #7047, I am wondering should we add it into canonical attribute list to suppress the warning.

Reply:
I caught some bugs thanks to the warnings in the tests.